File: app/src/main/java/com/sliide/news/network/model/NewsItem.java
package com.sliide.news.network.model;

import com.google.gson.annotations.SerializedName;

public class NewsItem{

    @SerializedName("summary")
    public String summary;

    public String getImageUrl() {
        if(images != null && images.mainImage != null && images.mainImage.url != null) {
            return images.mainImage.url;
        }
        return "";
    }

    @SerializedName("images")
    public Image images;

    private class Image {
        @SerializedName("mainImage")
        public MainImage mainImage;

        private class MainImage {
            @SerializedName("url")
            public String url;
        }
    }
}

File: app/src/main/java/com/sliide/news/network/model/NewsItem.kt
package com.sliide.news.network.model

import com.google.gson.annotations.SerializedName

data class NewsItem(
    @SerializedName("title")
    val title: String? = null,

    @SerializedName("description")
    val summary: String? = null,

    @SerializedName("image_url")
    val imageUrl: String? = null,

    @SerializedName("content")
    val content: String? = null,

    @SerializedName("pubDate")
    val pubDate: String? = null,

    @SerializedName("source_id")
    val sourceId: String? = null,

    @SerializedName("link")
    val link: String? = null
)